go/kbfs/libfs/root_fs.go

```diff
@@ -0,0 +1,173 @@
+}
+
+var _ billy.Filesystem = (*RootFS)(nil)
+
+///// Read-only functions:
+
+var rootWrappedNodeNames = map[string]bool{
+	StatusFileName: true,
+}
+
+// Open implements the billy.Filesystem interface for RootFS.
+func (rfs *RootFS) Open(filename string) (f billy.File, err error) {
+	if !rootWrappedNodeNames[filename] {
+		// In particular, this FS doesn't let you open the folderlist
+		// directories or anything in them.
+		return nil, os.ErrNotExist
+	}
+
+	switch filename {
+	case StatusFileName:
+		return &wrappedReadFile{
+			name: StatusFileName,
+			reader: func(ctx context.Context) ([]byte, time.Time, error) {
+				return GetEncodedStatus(ctx, rfs.config)
+			},
+			log: rfs.log,
+		}, nil
+	default:
+		panic(fmt.Sprintf("Name %s was in map, but not in switch", filename))
+	}
+}
+
+// OpenFile implements the billy.Filesystem interface for RootFS.
+func (rfs *RootFS) OpenFile(filename string, flag int, _ os.FileMode) (
+	f billy.File, err error) {
+	if flag&os.O_CREATE != 0 {
+		return nil, errors.New("RootFS can't create files")
+	}
+
+	return rfs.Open(filename)
+}
+
+// Lstat implements the billy.Filesystem interface for RootFS.
+func (rfs *RootFS) Lstat(filename string) (fi os.FileInfo, err error) {
+	if filename == "" {
+		filename = "."
+	}
+	if filename == "." {
+		return &wrappedReadFileInfo{
+			"keybase", 0, rfs.config.Clock().Now(), true}, nil
+	}
+	if !rootWrappedNodeNames[filename] {
+		return nil, os.ErrNotExist
+	}
+
+	switch filename {
+	case StatusFileName:
+		wrf := &wrappedReadFile{
+			name: StatusFileName,
+			reader: func(ctx context.Context) ([]byte, time.Time, error) {
+				return GetEncodedStatus(ctx, rfs.config)
+			},
+			log: rfs.log,
+		}
+		return wrf.GetInfo(), nil
+	default:
+		panic(fmt.Sprintf("Name %s was in map, but not in switch", filename))
+	}
+}
+
+// Stat implements the billy.Filesystem interface for RootFS.
+func (rfs *RootFS) Stat(filename string) (fi os.FileInfo, err error) {
+	return rfs.Lstat(filename)
+}
+
+// Join implements the billy.Filesystem interface for RootFS.
+func (rfs *RootFS) Join(elem ...string) string {
+	return path.Clean(path.Join(elem...))
+}
+
+// ReadDir implements the billy.Filesystem interface for RootFS.
+func (rfs *RootFS) ReadDir(p string) (fis []os.FileInfo, err error) {
+	if p == "" {
+		p = "."
+	}
+
+	if p != "." {
+		return nil, os.ErrNotExist
+	}
+
+	now := rfs.config.Clock().Now()
+	return []os.FileInfo{
+		&wrappedReadFileInfo{"private", 0, now, true},
+		&wrappedReadFileInfo{"public", 0, now, true},
+		&wrappedReadFileInfo{"team", 0, now, true},
+	}, nil
+}
+
+// Readlink implements the billy.Filesystem interface for RootFS.
+func (rfs *RootFS) Readlink(_ string) (target string, err error) {
+	return "", errors.New("RootFS cannot read links")
+}
+
+// Chroot implements the billy.Filesystem interface for RootFS.
+func (rfs *RootFS) Chroot(_ string) (newFS billy.Filesystem, err error) {
+	return nil, errors.New("RootFS cannot read links")
```

Comment:
`RootFS cannot chroot` (but also, why not?)

Reply:
Because it could only chroot up from the root FS into say a folder list like `/private`, or all the way into the TLF, and I don't see a need to implement all that right now. Seems better to just prevent it.